app/src/main/java/com/pedroid/qrcodecompose/androidapp/features/generate/navigation/customize/format/QRCodeFormatSelectionNavigation.kt

```diff
@@ -27,7 +26,8 @@ fun NavGraphBuilder.generateQRCodeSelectFormatRoute(
     largeScreen: Boolean = false,
 ) {
     composable<SelectFormatRoute>(
-        typeMap = mapOf(typeOf<QRCodeComposeXFormat>() to NavType.ofQRCodeComposeXFormat()),
+        // TODO remove this type map once https://issuetracker.google.com/issues/346475493 is addressed
+        typeMap = mapOf(typeOf<QRCodeComposeXFormat>() to NavType.EnumType(QRCodeComposeXFormat::class.java)),
```

Comment:
This is the actual fix; The other changes are other minor improvements I had made in the attempt of fixing the issue, but are still nice-to-have so I left them